File: csw-location/csw-location-api/jvm/src/main/scala/csw/location/api/javadsl/JConnectionType.scala
package csw.location.api.javadsl

import csw.location.models.ConnectionType

/**
 * Helper class for Java to get the handle of `ConnectionType` which is fundamental to LocationService library
 */
object JConnectionType {

  /**
   * Used to define an Akka connection
   */
  val AkkaType: ConnectionType = ConnectionType.AkkaType

  /**
   * Used to define a TCP connection
   */
  val TcpType: ConnectionType = ConnectionType.TcpType

  /**
   * Used to define a HTTP connection
   */
  val HttpType: ConnectionType = ConnectionType.HttpType
}

File: csw-location/csw-location-api/jvm/src/main/java/csw/location/api/javadsl/JConnectionType.java
/*
 * Copyright (c) 2022 Thirty Meter Telescope International Observatory
 * SPDX-License-Identifier: Apache-2.0
 */

package csw.location.api.javadsl;


import csw.location.api.models.ConnectionType;

/**
 * Helper class for Java to get the handle of `ConnectionType` which is fundamental to LocationService library
 */
public interface JConnectionType {
    /**
     * Used to define an Akka connection
     */
    ConnectionType AkkaType = ConnectionType.AkkaType$.MODULE$;

    /**
     * Used to define a TCP connection
     */
    ConnectionType TcpType = ConnectionType.TcpType$.MODULE$;

    /**
     * Used to define a HTTP connection
     */
    ConnectionType HttpType = ConnectionType.HttpType$.MODULE$;
}